de.metas.acct.base/src/main/sql/postgresql/ddl/functions/BusinessPartnerAccountSheetReport.sql

```diff
@@ -0,0 +1,250 @@
+
+
+    --
+    -- Update the beginning and end balances with the initial "Open Invoice Amount to Date"
+    UPDATE temp_BusinessPartnerAccountSheetReport
+    SET beginningBalance = t.OpenInvoiceAmountToDate,
+        endingBalance    = t.OpenInvoiceAmountToDate
+    FROM (
+             SELECT getBPOpenAmtToDate(p_ad_client_id,
+                                       p_ad_org_id,
+                                       (p_dateFrom - INTERVAL '1 days')::date,
+                                       p_c_bpartner_id,
+                                       (SELECT c.c_currency_id
+                                        FROM c_currency c
+                                                 INNER JOIN c_acctschema accts ON c.c_currency_id = accts.c_currency_id
+                                                 INNER JOIN ad_clientinfo ac ON accts.c_acctschema_id = ac.c_acctschema1_id
+                                        LIMIT 1),
+                                       'Y'::text,
+                                       p_isSoTrx) OpenInvoiceAmountToDate
+         ) t;
+
+    GET DIAGNOSTICS v_temp = ROW_COUNT;
+    v_time := logDebug('Update beginning and end balance with "BP Open Invoices Amount to Date"', v_time);
+
+
+    --
+    -- Compute rolling sum
+    WITH endingBalanceSum AS
+             (
+                 SELECT --
+                        t.rowid,
+                        t.endingBalance
+                            + sum(t.amount) OVER ( ORDER BY t.dateacct, t.created, t.documentno ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW ) endingBalance,
+                        t.amount                                                                                                                   currentAmount
+                 FROM temp_BusinessPartnerAccountSheetReport t
+             ),
+         finalData AS
+             (
+                 SELECT --
+                        ebs.rowid,
+                        ebs.endingBalance,
+                        ebs.endingBalance - ebs.currentAmount beginningBalance
+                 FROM endingBalanceSum ebs
+             )
+    UPDATE temp_BusinessPartnerAccountSheetReport t
+    SET endingBalance    = d.endingBalance,
+        beginningBalance = d.beginningBalance
+    FROM finalData d
+    WHERE t.rowid = d.rowid;
+
+    v_time := logDebug('finished calculating rolling sum', v_time);
+
+
+    --
+    -- return the data
+    RETURN QUERY SELECT --
+                        t.dateAcct,
+                        t.doctype,
+                        t.documentno,
+                        t.beginningBalance,
+                        t.amount,
+                        t.endingBalance,
+                        t.targetCurrencyCode,
+                        t.description
+                 FROM temp_BusinessPartnerAccountSheetReport t
+                 ORDER BY t.dateacct, t.created, t.documentno;
+END;
+$BODY$
+    LANGUAGE plpgsql
+    VOLATILE;
+
+
+/*
+How to run:
```

Comment:
please add this as a comment to the function; it's more accessible; feel free to add more infos. you can also add the public issue's URL there

Reply:
done.